GCC-15 compatibility

GCC-15 just (ten days ago) merged some new template correctness checks:

https://trofi.github.io/posts/322-gcc-15-template-checking-improvements.html

One of our CI machines already detected a failure in the GAP semigroups package that can be traced back to `sims1.tpp` in libsemigroups: https://bugs.gentoo.org/936564

I've copy/pasted the error below for convenience:

```
In file included from /usr/include/libsemigroups/sims1.hpp:1269,
                 from src/pkg.cpp:59:
/usr/include/libsemigroups/sims1.tpp: In member function libsemigroups::Sims1<T>::iterator_base& libsemigroups::Sims1<T>::iterator_base::operator=(libsemigroups::Sims1<T>::iterator_base&&):
/usr/include/libsemigroups/sims1.tpp:622:39: error: class libsemigroups::Sims1<T>::iterator_base has no member named long_rules
  622 |     _longs           = std::move(that.long_rules());
      |                                       ^~~~~~~~~~
```

Not an immediate problem but if you want to get ahead of the GCC-15 release, it's gonna break things...

I managed to patch away my gcc-15 issue, and good thing I did. There is actually still a build failure in the test suite:

```
tests/test-todd-coxeter.cpp: In function 'void libsemigroups::CATCH2_INTERNAL_TEST_65()':
tests/test-todd-coxeter.cpp:1445:21: error: call of overloaded 'report_every(int)' is ambiguous
 1445 |     tc2.report_every(1);
      |     ~~~~~~~~~~~~~~~~^~~
tests/test-todd-coxeter.cpp:1445:21: note: there are 4 candidates
In file included from /var/lib/portage/tmp/portage/sci-libs/libsemigroups-3.0.2/work/libsemigroups-3.0.2/include/libsemigroups/todd-coxeter-class.hpp:44,
                 from /var/lib/portage/tmp/portage/sci-libs/libsemigroups-3.0.2/work/libsemigroups-3.0.2/include/libsemigroups/to-froidure-pin.hpp:31,
                 from tests/test-todd-coxeter.cpp:35:
/var/lib/portage/tmp/portage/sci-libs/libsemigroups-3.0.2/work/libsemigroups-3.0.2/include/libsemigroups/detail/todd-coxeter-impl.hpp:510:12: note: candidate 1: 'void libsemigroups::detail::ToddCoxeterImpl::report_every(T) [with T = int]'
  510 |       void report_every(T val) {
      |            ^~~~~~~~~~~~
In file included from /var/lib/portage/tmp/portage/sci-libs/libsemigroups-3.0.2/work/libsemigroups-3.0.2/include/libsemigroups/froidure-pin-base.hpp:32,
                 from /var/lib/portage/tmp/portage/sci-libs/libsemigroups-3.0.2/work/libsemigroups-3.0.2/include/libsemigroups/froidure-pin.hpp:35,
                 from tests/test-todd-coxeter.cpp:29:
/var/lib/portage/tmp/portage/sci-libs/libsemigroups-3.0.2/work/libsemigroups-3.0.2/include/libsemigroups/runner.hpp:209:15: note: candidate 2: 'libsemigroups::Reporter& libsemigroups::Reporter::report_every(Time) [with Time = int]'
  209 |     Reporter& report_every(Time t) noexcept {
      |               ^~~~~~~~~~~~
```

--- a/include/libsemigroups/detail/todd-coxeter-impl.hpp
+++ b/include/libsemigroups/detail/todd-coxeter-impl.hpp
@@ -505,13 +505,16 @@ namespace libsemigroups {
       ////////////////////////////////////////////////////////////////////////
 
 #ifndef LIBSEMIGROUPS_PARSED_BY_DOXYGEN
-      // This is documented in Runner, so we don't duplicate the doc here.
-      template <typename T>
-      void report_every(T val) {
+      // This is documented in Reporter, so we don't duplicate the doc here.
+      template <typename Time>
+      void report_every(Time val) {
         detail::CongruenceCommon::report_every(val);
         _word_graph.report_every(val);
       }
-      using Reporter::report_every;
+
+      [[nodiscard]] nanoseconds report_every() const noexcept {
+        return Reporter::report_every();
+      }
 #endif
 
       //! \ingroup todd_coxeter_class_settings_group

